homeassistant/components/hassio/__init__.py

```diff
@@ -704,6 +709,19 @@ async def async_handle_core_service(call: ServiceCall) -> None:
     # Init add-on ingress panels
     await async_setup_addon_panel(hass, hassio)
 
+    # Setup hardware integration for the detected board type
+    async def _async_setup_hardware_integration(hass):
```

Comment:
Should this be specific for OS/Supervisor based installations only? Hardware can be part of others as well, right?

Reply:
If we only support board detection it will be limited to supervisor based installations.

Reply:
We can start with supervisor in the MVP, and consider extending support later